service/history/timer_queue_standby_task_executor.go

```diff
@@ -317,16 +317,15 @@ func (t *timerQueueStandbyTaskExecutor) executeActivityRetryTimerTask(
 			return nil, err
 		}
 
-		if activityInfo.Attempt > task.Attempt {
-			return nil, nil
-		}
-
-		if activityInfo.Stamp != task.Stamp {
-			// this retry task is from old Stamp. In this case we should ignore it
-			return nil, nil
-		}
-
-		if activityInfo.StartedEventId != common.EmptyEventID {
+		// we ignore retry timer task if:
+		// * this retry task is from old Stamp.
+		// * attempts is not the same as recorded in activity info.
+		// * activity is already started.
+		// * activity is paused.
+		if activityInfo.Attempt > task.Attempt ||
+			activityInfo.Stamp != task.Stamp ||
+			activityInfo.StartedEventId != common.EmptyEventID ||
+			activityInfo.Paused {
```

Comment:
Wouldn't the stamp be updated in the standby cluster if the activity is paused? Why not use the same mechanism?

Reply:
.Puased check is not needed